build-logic/uber-plugins/src/main/kotlin/gradlebuild.java-library.gradle.kts

```diff
@@ -33,3 +33,9 @@ plugins {
     id("gradlebuild.ci-reporting") // CI: Prepare reports to be uploaded to TeamCity
     id("gradlebuild.configure-ci-artifacts") // CI: Prepare reports to be uploaded to TeamCity
 }
+
+dependencies {
+    if (project.name != "base-annotations") {
+        compileOnly(project(":base-annotations"))
+    }
+}
```

Comment:
I think this is too broad -- I'm not sure projects like `performance` need this dependency. It should probably be added to each `build.gradle.kts` as needed.